napari/_qt/qt_viewer.py

```diff
@@ -482,6 +483,70 @@ def _create_performance_dock_widget(self):
             )
         return None
 
+    def _weakref_if_possible(self, obj):
+        """Create a weakref to obj.
+
+        Cannot create weakrefs to many Python built-in datatypes such as list, dict, str.
+
+        From https://docs.python.org/3/library/weakref.html: "Objects which support weak
+        references include class instances, functions written in Python (but not in C),
+        instance methods, sets, frozensets, some file objects, generators, type objects,
+        sockets, arrays, deques, regular expression pattern objects, and code objects."
+
+        Args:
+            obj (Object): object
+
+        Returns:
+            weakref|Object: Returns a weakref if possible
+        """
+        try:
+            newref = ref(obj)
+        except TypeError:
+            newref = obj
+        return newref
+
+    def _unwrap_if_weakref(self, value):
+        """Returns value or if that is weakref the object referenced by value.
+
+        Args:
+            value (Object|weakref): Can be a weakref
+
+        Returns:
+            Object|None: Returns object or None if weakref is dead.
```

Comment:
@davidpross can you use numpydoc formatting here? 🙏 (and above). Otherwise, nice, thanks! 😊

Reply:
Updated docstrings, thanks for all the help with these changes.